lib/ipip.c

```diff
@@ -33,7 +33,7 @@ encapsulate(struct rte_mbuf *pkt, uint8_t priority, struct ip_flow *flow)
 	if (flow->proto == ETHER_TYPE_IPv4) {
 		/* Allocate space for outer IPv4 header. */
 		outer_ip4hdr = (struct ipv4_hdr *)rte_pktmbuf_prepend(pkt,
-			sizeof(struct ipv4_hdr) - sizeof(struct ether_hdr));
+			sizeof(struct ipv4_hdr));
```

Comment:
This code writes the new, outer IPv4 header at the start of the packet buffer, but the outer IPv4 header should not be at the beginning of the buffer -- the Ethernet header should be first. So this code fills in the IPv4 header at the beginning, then returns and writes the Ethernet header over many of the same bytes.